plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerConnector.java

```diff
@@ -765,28 +706,72 @@ public void putResource(String container, String targetPath, InputStream sourceS
 
         List<Pair<String, ?>> headers = Arrays.asList(Pair.of("Content-Type", ExtMediaType.APPLICATION_X_TAR),
                                                       Pair.of("Content-Length", length));
-        DockerConnection connection = null;
-        try (InputStream tarStream = new BufferedInputStream(new FileInputStream(tarFile))) {
-            connection = openConnection(dockerDaemonUri).method("PUT")
-                                                        .path(String.format("/containers/%s/archive?path=%s&noOverwriteDirNonDir=%d",
-                                                                            container, targetPath, overwrite ? 0 : 1))
-                                                        .headers(headers)
-                                                        .entity(tarStream);
+        try (InputStream tarStream = new BufferedInputStream(new FileInputStream(tarFile));
+             DockerConnection connection = openConnection(dockerDaemonUri).method("PUT")
+                                                                          .path("/containers/" + container + "/archive")
+                                                                          .query("path", targetPath)
+                                                                          .query("noOverwriteDirNonDir", noOverwriteDirNonDir ? 0 : 1)
+                                                                          .headers(headers)
+                                                                          .entity(tarStream)) {
             final DockerResponse response = connection.request();
             final int status = response.getStatus();
             if (status != OK.getStatusCode()) {
-                final String m = CharStreams.toString(new InputStreamReader(response.getInputStream()));
-                throw new DockerException(String.format("Error response from docker API, status: %d, message: %s", status, m), status);
+                throw new DockerException(getDockerExceptionMessage(response), status);
             }
         } finally {
-            if (connection != null) {
-                connection.close();
-            }
             FileCleaner.addFile(tarFile);
         }
     }
 
     /**
+     * Get docker events.
```

Comment:
Then its okay. As for me it is just better to separate _Summary_ from _Description_ with 1 blank line.